# Analysis of Micromonas psr1-like gene in TARA Metatranscriptomic Data 

This repository is associated with the publication: 

**A phosphate starvation response gene (*psr*1-like) is present and expressed in *Micromonas pusilla* 
and other marine algae and may coordinate a metabolic response to phosphorus deficiency** 
Cara L. Fiore, Harriet Alexander, Melissa C. Kido Soule, Elizabeth B. Kujawinski [bioarxiv](link). 

## Download and running

This repository contains the series of Jupyter Notebooks used in the analysis of the presence/absence of Micromonas-specific *psr*1-like genes 
in the TARA Oceans Eukaryotic gene catalogue [(Carradec et al. 2018)](https://www.nature.com/articles/s41467-017-02342-1). 

First, clone the directory and `cd` into it: 

```
clone https://github.com/AlexanderLabWHOI/micromonas-psr1-tara.git
cd micromonas-psr1-tara
```

To run the notebook, have [anaconda](https://www.anaconda.com/download/) installed and create a conda environment in which to run
the notebooks: 

```
conda env create -f environment.yml     
source activate tara_plot     
jupyter notebook
```

This will launch a jupyter enivonrment in which all jupyter notebooks might be run. 

## Contents
- `dask-parse.ipynb` : Outlines the dask-based computations used to manipulate and analyze the large
gene-sample table `metatranscriptomic_occurrences.tsv` (which is a ~120Gb tab delimited file). To run this notebook you will 
need to download a copy of `metatranscriptomic_occurrences.tsv` locally and adjust the dask-jobqueue parameters to match that 
of your own HPC. 
- `PSR1-final-figures.ipynb` : Runs analysis for *psr*1 abundance in the transcriptomic data, normalizes data to total Micromonas abundance, 
and plots transcript data against environmental parameters. 

Note: All data files required for the running of the `PSR1-final-figures.ipynb` are provided in `input`-- including those which 
are generated by `dask-parse.ipynb`. 
